File: backend/app/schemas/audit_log.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class AuditLogRead(BaseModel):
    """返回给管理员查看的审计日志。"""

    id: int
    actor_user_id: int | None
    actor_username_snapshot: str | None
    actor_role_snapshot: str | None
    actor_source: str
    action: str
    target_type: str
    target_id: str
    detail_json: dict[str, Any] | None
    result: str
    ip_address: str | None
    request_id: str | None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class AuditLogListResponse(BaseModel):
    """审计日志分页响应。"""

    items: list[AuditLogRead]
    total: int
    page: int
    page_size: int

File: frontend/src/types/auditLog.ts
export interface AuditLog {
  id: number
  actor_user_id: number | null
  actor_username_snapshot: string | null
  actor_role_snapshot: string | null
  actor_source: string
  action: string
  target_type: string
  target_id: string
  detail_json: Record<string, unknown> | null
  result: string
  ip_address: string | null
  request_id: string | null
  created_at: string
}

export interface AuditLogListResponse {
  items: AuditLog[]
  total: number
  page: number
  page_size: number
}
